sklearn/tests/test_grid_search.py

```diff
@@ -217,23 +217,6 @@ def test_grid_search_score_method():
                                               scoring='roc_auc').fit(X, y)
     search_auc = GridSearchCV(clf, grid, scoring='roc_auc').fit(X, y)
 
-    # Check warning only occurs in situation where behavior changed:
-    # estimator requires score method to compete with scoring parameter
-    score_no_scoring = assert_no_warnings(search_no_scoring.score, X, y)
-    score_accuracy = assert_warns(ChangedBehaviorWarning,
-                                  search_accuracy.score, X, y)
-    score_no_score_auc = assert_no_warnings(search_no_score_method_auc.score,
-                                            X, y)
-    score_auc = assert_warns(ChangedBehaviorWarning,
-                             search_auc.score, X, y)
-    # ensure the test is sane
```

Comment:
There are two places where that checks if `ChangedBehaviorWarning` is raised or not. If I remove one, the other raises error. I had two options, one is the change that this PR now proposes, the other being the following diff (with respect to this PR): 

```
diff --git a/sklearn/tests/test_grid_search.py b/sklearn/tests/test_grid_search.py
index d4f5664..5eaeba7 100644
--- a/sklearn/tests/test_grid_search.py
+++ b/sklearn/tests/test_grid_search.py
@@ -217,6 +217,20 @@ def test_grid_search_score_method():
                                               scoring='roc_auc').fit(X, y)
     search_auc = GridSearchCV(clf, grid, scoring='roc_auc').fit(X, y)
 
+    # Check warning only occurs in situation where behavior changed:
+    # estimator requires score method to compete with scoring parameter
+    score_no_scoring = assert_no_warnings(search_no_scoring.score, X, y)
+    score_accuracy = assert_no_warnings(ChangedBehaviorWarning,
+                                  search_accuracy.score, X, y)
+    score_no_score_auc = assert_no_warnings(search_no_score_method_auc.score,
+                                            X, y)
+    score_auc = assert_no_warnings(ChangedBehaviorWarning,
+                             search_auc.score, X, y)
+    # ensure the test is sane
+    assert_true(score_auc > 1.0)
+    assert_true(score_accuracy > 1.0)
+    assert_not_equal(score_auc, score_accuracy)
+
 ```